### PR TITLE
ISSITES-5840 Add Search Functionality to Public Sites Nav

### PR DESCRIPTION
**Pull request by liaoethan**

### CODE REVIEW

**Comment by liaoethan** [comment]
ci:test:sf

**Comment by allen-ziegenfus** [comment]
@liaoethan  Can you update this to not redefine the replaceAll function? It should be possible to do a case insensitive regex replace without that.

File: modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.js [hunk 1]
```diff
@@ -60,4 +60,177 @@ window.addEventListener('load', () => {
...
+
+		if (data && data.items && data.items[0]) {
+			const myjson = JSON.parse(JSON.stringify(data.items[0]));
+			if (myjson) {
+				searchSuggestionsResult.innerHTML = '';
+
+				for (const suggestion of myjson.suggestions) {
+					searchSuggestionsResult.classList.remove('d-none');
+
+					const newSuggestion = document.createElement('div');
```
Comment (allen-ziegenfus) [inline modules/apps/site-initializer/site-initializer-liferay-learn/src/main/resources/site-initializer/fragments/group/public-sites-navigation/fragments/public-sites-navigation/index.js:134]: @liaoethan  shouldn't this be an anchor link?

**Comment by liaoethan** [comment]
Hi @allen-ziegenfus, I made the two changes that we discussed. Let me know if I need to do anything additional, thanks!

**Comment by allen-ziegenfus** [comment]
merged here: https://github.com/allen-ziegenfus/liferay-portal/pull/73

I simplified a bunch of things @liaoethan  since I noticed there was some duplicated code.

**Comment by liaoethan** [comment]
Thanks @allen-ziegenfus !

**Pull request closed without merging**